crates/turborepo-lib/src/turbo_json.rs

```diff
@@ -0,0 +1,31 @@
+use std::{fs, path::Path};
+
+use anyhow::Result;
+use semver::{Version, VersionReq};
+use serde::Deserialize;
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct TurboJson {
+    pub turbo_version: String,
+}
+
+pub fn read_turbo_json(path: &Path) -> Result<TurboJson> {
+    let turbo_json_string = fs::read_to_string(path)?;
+    let turbo_json = serde_json::from_str(&turbo_json_string)?;
+
+    Ok(turbo_json)
+}
+
+impl TurboJson {
+    pub fn check_version(&self, version: &str) -> Result<bool> {
+        let version = Version::parse(version)?;
+
+        if self.turbo_version.is_empty() {
+            return Ok(true);
+        }
+
+        let version_request = VersionReq::parse(&self.turbo_version)?;
```

Comment:
This is a Cargo-style SemVer comparison string, which is _slightly_ different from what `node` uses.

We can't choose "both", so we should make an active decision about which pattern we want to support.